### DISCUSSION TITLE
Getting facted results for multiple fields

### DISCUSSION THREAD

**Discussion by chris-fellows** [ravendb/ravendb#15351] [Q&A]
RavenDB 5.4.5 (Using Dot Net Core)
I have a map/reduce index that aggregates order counts by 1 hour periods. It has fields PeriodStart (DateTime - Start of 1 hr period), StoreId (String), ProductCategoryId (String), ProductId (String) & PeriodOrderCount (Aggregated count).

For each 1 hr within the last 48 hours then I want to get PeriodStart + StoreId + ProductCategoryId + sum(PeriodOrderCount). There are a lot of map reduce items and summing them up client side is too slow. And I don't want to create another map/reduce index that aggregates the same fields except for ProductId.

How would I accomplish it with facets? Do I have to loop through every combination of StoreId + ProductCategoryId and get the faceted result or can RavenDB do that for me?

**Comment by ayende** [comment]
In your index, output a single field that has the concatenation of all those values.
Then you can run a facet on this.

```
// map
from p in docs.Orders
select new 
{ 
///   ....
}

// reduce
from r in results 
group r by new { r.Date, .. } into g
select new 
{
   FieldToFacetAggregation = g.Key.PriodStart +"/" + g.Key.ProductCategoryId +"/" ...,
}
```

And then in your query, do:

```
from index index_name
select facet(FieldToFacetAggregation, sum(PreiodOrderCount))
```
